### PR TITLE
[Feature] add tensorboard support

### PR DESCRIPTION
**Pull request by rbao2018**
# Add TensorBoard Support to Slime

## 🎯 Overview

This PR adds comprehensive TensorBoard support to Slime alongside the existing Weights & Biases (wandb) integration, providing users with more flexibility in choosing their preferred logging and visualization tools for training monitoring.

### 🔧 Unified Logging Interface
- **New unified logger**: Introduces `slime.utils.logger_utils.UnifiedLogger` that seamlessly supports both wandb and TensorBoard
- **Backward compatibility**: All existing wandb functionality remains unchanged
- **Minimal code changes**: Uses a single `log_metric()` function for both logging backends
- **Clean separation**: TensorBoard and wandb can be used independently or together

### 🎛️ Command Line Arguments
Added new arguments for TensorBoard configuration:
```bash
--use-tensorboard              # Enable TensorBoard logging
--tensorboard-log-dir PATH     # Directory for TensorBoard logs (default: ./runs)
```

### 📊 Comprehensive Metric Logging
TensorBoard integration captures all the same metrics as wandb:
- **Training metrics**: loss, learning rate, gradient norms, KL divergence, etc.
- **Rollout metrics**: rewards, response lengths, generation performance
- **Evaluation metrics**: accuracy, pass rates, multi-turn conversation metrics
- **Performance metrics**: throughput, memory usage, timing information

### Core Infrastructure
- **`slime/utils/logger_utils.py`** (NEW): Unified logging interface with TensorBoard support
- **`slime/utils/arguments.py`**: Added TensorBoard command-line arguments
- **`slime/utils/wandb_utils.py`**: Integrated unified logger initialization
- **`requirements.txt`**: Added `tensorboard` dependency

### Updated Logging Calls
Replaced `wandb.log()` calls with unified `log_metric()` across:
- **`slime/ray/buffer.py`**: Evaluation and performance logging
- **`slime/backends/megatron_utils/model.py`**: Training step logging
- **`slime/backends/megatron_utils/data.py`**: Rollout data and performance logging
- **`slime/backends/fsdp_utils/actor.py`**: FSDP training and rollout logging

### TensorBoard Only
```bash
python train.py \
    --use-tensorboard \
    --tensorboard-log-dir ./tensorboard_logs \
    [other training arguments]
```

### Both TensorBoard and Wandb
```bash
python train.py \
    --use-wandb \
    --use-tensorboard \
    --tensorboard-log-dir ./tensorboard_logs \
    [other training arguments]
```

### Wandb Only (unchanged)
```bash
python train.py \
    --use-wandb \
    [other training arguments]
```

### CODE REVIEW

**Comment by fy1214** [comment]
can use use-pytorch-profiler argument which is already exist

**Comment by rbao2018** [comment]
> can use use-pytorch-profiler argument which is already exist

I checked and searched the main branch of the current repository, but I didn't find the "use-pytorch-profile" argument you mentioned. Could you please specify it more precisely?

**Comment by fy1214** [comment]
> > can use use-pytorch-profiler argument which is already exist
> 
> I checked and searched the main branch of the current repository, but I didn't find the "use-pytorch-profile" argument you mentioned. Could you please specify it more precisely?

the argument is set in megatron argument parser but never use in slime
so i use this argument in slime/backends/megatron_utils/actor.py: 95 line

**Comment by rbao2018** [comment]
> > > can use use-pytorch-profiler argument which is already exist
> > 
> > 
> > I checked and searched the main branch of the current repository, but I didn't find the "use-pytorch-profile" argument you mentioned. Could you please specify it more precisely?
> 
> the argument is set in megatron argument parser but never use in slime so i use this argument in slime/backends/megatron_utils/actor.py: 95 line

Ah, you're right. Additionally, I found that the ```--tensorboard-dir``` is also defined in the megatron argument, so maybe I can make use of this parameter and then refactor my current implementation.

**Comment by fy1214** [comment]
> > > > can use use-pytorch-profiler argument which is already exist
> > > 
> > > 
> > > I checked and searched the main branch of the current repository, but I didn't find the "use-pytorch-profile" argument you mentioned. Could you please specify it more precisely?
> > 
> > 
> > the argument is set in megatron argument parser but never use in slime so i use this argument in slime/backends/megatron_utils/actor.py: 95 line
> 
> Ah, you're right. Additionally, I found that the `--tensorboard-dir` is also defined in the megatron argument, so maybe I can make use of this parameter and then refactor my current implementation.

sure

**Comment by zhuzilin** [comment]
wow, this is amazing! Thank you so much!

**Comment by rbao2018** [comment]
> wow, this is amazing! Thank you so much!

I think my current PR still needs a lot of modifications. In particular, some unnecessary parameters can be simplified as much as possible. I will refine this PR and reopen it.
